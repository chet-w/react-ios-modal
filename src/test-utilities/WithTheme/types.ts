import { ReactElement } from "react";

export interface WithThemeProps {
  children?: ReactElement;
}