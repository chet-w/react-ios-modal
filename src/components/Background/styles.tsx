import styled from "styled-components";
import { motion } from "framer-motion";

export const Background = styled(motion.div)`
  height: 100vh;
  width: 100vw;
  margin: 0;
  padding: 0;
  position: absolute;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: flex-start;
  z-index: 10000;
`;

export const variants = {
  closed: {
    backdropFilter: "none",
    background: "rgba(255, 255, 255, 0)"
  },
  open: {
    backdropFilter: "blur(5px)",
    background: "rgba(255, 255, 255, 0.5)"
  }
};

export const mobileVariants = {
  closed: {
    background: "rgba(0, 0, 0, 0)"
  },
  open: {
    background: "rgba(0, 0, 0, 0.5)"
  }
};
