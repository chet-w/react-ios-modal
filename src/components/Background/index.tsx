import * as S from "./styles";
import { BackgroundProps } from "./types";

const Background = (backgroundProps: BackgroundProps) => {
  return (
    <S.Background
      {...backgroundProps}
      variants={S.animations}
      initial="closed"
      animate="open"
      exit="closed"
    />
  );
};

export default Background;
