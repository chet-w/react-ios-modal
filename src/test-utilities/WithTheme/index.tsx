import { ThemeProvider } from "styled-components";
import { modalTheme } from "../../components/Theme";
import { WithThemeProps } from "./types";

const WithTheme = ({ children }: WithThemeProps) => (
  <ThemeProvider theme={modalTheme}>{children}</ThemeProvider>
);

export default WithTheme;
