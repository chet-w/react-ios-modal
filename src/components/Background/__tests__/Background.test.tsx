import { render } from "@testing-library/react";
import { AnimatePresence } from "framer-motion";
import Background from "../";
import WithTheme from "../../../test-utilities/WithTheme";

const renderBackground = (props = {}) => {
  const { container } = render(
    <WithTheme>
      <AnimatePresence>
        <Background {...props} />
      </AnimatePresence>
    </WithTheme>
  );
  return container;
};

describe("Background", () => {
  it("renders successfully", () => {
    const background = renderBackground();
    expect(background).toBeTruthy();
  });

  describe("when it's a mobile device", () => {
    it("has the correct mobile styling", () => {
      const background = renderBackground({ isMobile: true })
        .firstElementChild as HTMLElement;
      const styles = getComputedStyle(background);
      expect(styles).toMatchSnapshot();
    });
  });

  describe("when it's not a mobile device", () => {
    it("has the correct non-mobile styling", () => {
      const background = renderBackground({ isMobile: false })
        .firstElementChild as HTMLElement;
      const styles = getComputedStyle(background);
      expect(styles).toMatchSnapshot();
    });
  });
});
