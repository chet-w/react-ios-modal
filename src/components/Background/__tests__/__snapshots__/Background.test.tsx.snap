// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background when it's a mobile device has the correct mobile styling 1`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "height",
  "10": "z-index",
  "11": "overflow",
  "12": "visibility",
  "13": "background",
  "2": "width",
  "3": "margin",
  "4": "padding",
  "5": "position",
  "6": "top",
  "7": "left",
  "8": "justify-content",
  "9": "align-items",
  "_importants": Object {
    "align-items": "",
    "background": "",
    "display": "",
    "height": "",
    "justify-content": "",
    "left": "",
    "margin": "",
    "overflow": "",
    "padding": "",
    "position": "",
    "top": "",
    "visibility": undefined,
    "width": "",
    "z-index": "",
  },
  "_length": 14,
  "_onChange": [Function],
  "_values": Object {
    "align-items": "flex-start",
    "background": "rgba(0, 0, 0, 0)",
    "background-color": "rgba(0, 0, 0, 0)",
    "display": "flex",
    "height": "100%",
    "justify-content": "center",
    "left": "0px",
    "margin": "0px",
    "margin-bottom": "0px",
    "margin-left": "0px",
    "margin-right": "0px",
    "margin-top": "0px",
    "overflow": "hidden",
    "padding": "0px",
    "padding-bottom": "0px",
    "padding-left": "0px",
    "padding-right": "0px",
    "padding-top": "0px",
    "position": "absolute",
    "top": "0px",
    "visibility": "visible",
    "width": "100vw",
    "z-index": "10000",
  },
}
`;

exports[`Background when it's not a mobile device has the correct non-mobile styling 1`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "height",
  "10": "z-index",
  "11": "overflow",
  "12": "visibility",
  "13": "background",
  "2": "width",
  "3": "margin",
  "4": "padding",
  "5": "position",
  "6": "top",
  "7": "left",
  "8": "justify-content",
  "9": "align-items",
  "_importants": Object {
    "align-items": "",
    "background": "",
    "display": "",
    "height": "",
    "justify-content": "",
    "left": "",
    "margin": "",
    "overflow": "",
    "padding": "",
    "position": "",
    "top": "",
    "visibility": undefined,
    "width": "",
    "z-index": "",
  },
  "_length": 14,
  "_onChange": [Function],
  "_values": Object {
    "align-items": "flex-start",
    "background": "rgba(255, 255, 255, 0)",
    "background-color": "rgba(255, 255, 255, 0)",
    "display": "flex",
    "height": "100%",
    "justify-content": "center",
    "left": "0px",
    "margin": "0px",
    "margin-bottom": "0px",
    "margin-left": "0px",
    "margin-right": "0px",
    "margin-top": "0px",
    "overflow": "hidden",
    "padding": "0px",
    "padding-bottom": "0px",
    "padding-left": "0px",
    "padding-right": "0px",
    "padding-top": "0px",
    "position": "absolute",
    "top": "0px",
    "visibility": "visible",
    "width": "100vw",
    "z-index": "10000",
  },
}
`;
